swarm/tests/swarm_derive.rs

```diff
@@ -112,12 +110,8 @@ fn three_fields() {
         let _out_event: <Foo as NetworkBehaviour>::ToSwarm = unimplemented!();
         match _out_event {
             FooEvent::Ping(ping::Event { .. }) => {}
-            FooEvent::Identify(event) => {
-                let _: identify::Event = event;
-            }
-            FooEvent::Kad(event) => {
-                let _: libp2p_kad::Event = event;
-            }
+            FooEvent::Identify(_event) => {}
+            FooEvent::Kad(_event) => {}
```

Comment:
```suggestion
            FooEvent::Identify(_event) => {
                let _: identify::Event = _event;
            }
            FooEvent::Kad(_event) => {
                let _: libp2p_kad::Event = _event;
            }
```

This should also satisfy the lint, but maintains the type check.